api/prometheus/v1/api_test.go

```diff
@@ -829,6 +833,21 @@ func TestAPIClientDo(t *testing.T) {
 				Msg:  "inconsistent body for response code",
 			},
 		},
+		{
+			code: http.StatusOK,
+			response: &apiResponse{
+				Status:    "error",
+				Data:      json.RawMessage(`"test"`),
+				ErrorType: ErrTimeout,
+				Error:     "timed out",
+				Warnings:  []string{"a"},
+			},
+			expectedWarnings: []string{"a"},
```

Comment:
why do we need this separately.
can't we just have:
expectedErr: api.NewErrorAPI(nil, []string{"a"}),

and than the check can be just:
```
if test.expectedErr.Warning() != nil {
```

in the main block here :https://github.com/prometheus/client_golang/pull/562/files#diff-c03449ea3ae93306323741c88841a939R867